core/src/main/scala/org/apache/spark/sql/rapids/tool/store/TaskModel.scala

```diff
@@ -16,102 +16,155 @@
+  def sw_bytesWritten: Long = taskMetrics.getOrElse(SW_BYTES, 0L)
+  def sw_writeTime: Long = taskMetrics.getOrElse(SW_TIME, 0L)
+  def sw_recordsWritten: Long = taskMetrics.getOrElse(SW_RECORDS, 0L)
+  def input_bytesRead: Long = taskMetrics.getOrElse(INPUT_BYTES, 0L)
+  def input_recordsRead: Long = taskMetrics.getOrElse(INPUT_RECORDS, 0L)
+  def output_bytesWritten: Long = taskMetrics.getOrElse(OUTPUT_BYTES, 0L)
+  def output_recordsWritten: Long = taskMetrics.getOrElse(OUTPUT_RECORDS, 0L)
+}
 
 object TaskModel {
+  import LongMetrics._
+
   def apply(event: SparkListenerTaskEnd): TaskModel = {
+    val taskInfo = event.taskInfo
+    val metrics = event.taskMetrics
+    val shuffleRead = metrics.shuffleReadMetrics
+    val shuffleWrite = metrics.shuffleWriteMetrics
+    val input = metrics.inputMetrics
+    val output = metrics.outputMetrics
     val reason = event.reason match {
       case failed: TaskFailedReason =>
         failed.toErrorString
       case _ =>
         event.reason.toString
     }
 
-    TaskModel(
-      event.stageId,
-      event.stageAttemptId,
-      event.taskType,
-      reason,
-      event.taskInfo.taskId,
-      event.taskInfo.attemptNumber,
-      event.taskInfo.launchTime,
-      event.taskInfo.finishTime,
-      event.taskInfo.duration,
-      event.taskInfo.successful,
-      event.taskInfo.executorId,
-      event.taskInfo.host,
-      event.taskInfo.taskLocality.toString,
-      event.taskInfo.speculative,
-      event.taskInfo.gettingResultTime,
-      event.taskMetrics.executorDeserializeTime,
-      event.taskMetrics.executorDeserializeCpuTime, // nanoseconds
-      event.taskMetrics.executorRunTime, // milliseconds
-      event.taskMetrics.executorCpuTime, // nanoseconds
-      event.taskMetrics.peakExecutionMemory,
-      event.taskMetrics.resultSize,
-      event.taskMetrics.jvmGCTime,
-      event.taskMetrics.resultSerializationTime,
-      event.taskMetrics.memoryBytesSpilled,
-      event.taskMetrics.diskBytesSpilled,
-      event.taskMetrics.shuffleReadMetrics.remoteBlocksFetched,
-      event.taskMetrics.shuffleReadMetrics.localBlocksFetched,
-      event.taskMetrics.shuffleReadMetrics.fetchWaitTime,
-      event.taskMetrics.shuffleReadMetrics.remoteBytesRead,
-      event.taskMetrics.shuffleReadMetrics.remoteBytesReadToDisk,
-      event.taskMetrics.shuffleReadMetrics.localBytesRead,
-      event.taskMetrics.shuffleReadMetrics.totalBytesRead,
-      event.taskMetrics.shuffleWriteMetrics.bytesWritten,
-      event.taskMetrics.shuffleWriteMetrics.writeTime, // nanoseconds
-      event.taskMetrics.shuffleWriteMetrics.recordsWritten,
-      event.taskMetrics.inputMetrics.bytesRead,
-      event.taskMetrics.inputMetrics.recordsRead,
-      event.taskMetrics.outputMetrics.bytesWritten,
-      event.taskMetrics.outputMetrics.recordsWritten)
+
+    var taskMetrics = immutable.IntMap.empty[Long]
+
+    def storeIfNonZero(field: Int, value: Long): Unit =
```

Comment:
Can we include`{}` instead of inlined statements and a new line after the function definition?